# Урок 7. Непараметрические тесты

In [1]:
import numpy as np
import scipy.stats as stats

***
## Задание 1.

### Выбрать тест и проверить, есть  ли различия между выборками. Даны две независимые выборки. Не соблюдается условие нормальности. Сделайте вывод по результатам, полученным с помощью функции.

x1  380,420, 290 \
y1 140,360,200,900

Поскольку выборки 2, они независимы и не соблюдается условие нормальности, то выбираем критерий Манна-Уитни.

In [2]:
X = np.array([380, 420, 290])
Y = np.array([140, 360, 200, 900])

stats.mannwhitneyu(X,Y)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

Примем значение уровня значимости $\alpha=0.05$\
Т.к. p-value=0.62 больше 0.05, то гипотезу $H_0$ принимаем. Статистически значимой разницы между выборками нет.

***
## Задание 2.

### Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия? <br> 1е измерение до приема препарата: 150, 160, 165, 145, 155 <br> 2е измерение через 10 минут: 140, 155, 150,  130, 135 <br> 3е измерение через 30 минут: 130, 130, 120, 130, 125.

Поскольку исследуются множественные сравнения зависимых выборок (анализ повторных измерений), то выбираем критерий Фридмана.

In [3]:
izm1 = np.array([150, 160, 165, 145, 155])
izm2 = np.array([140, 155, 150, 130, 135])
izm3 = np.array([130, 130, 120, 130, 125])

stats.friedmanchisquare(izm1, izm2, izm3)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

Примем значение уровня значимости $\alpha=0.05$\
Т.к. p-value=0.008 меньше 0.05, то гипотезу $H_0$ отклоняем. Наблюдается статистически значимая разница между выборками.

***
## Задание 3.

### Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

Поскольку исследуются две зависимые выборки, то выбираем критерий Уилкоксона.

In [4]:
izm1 = np.array([150, 160, 165, 145, 155])
izm2 = np.array([140, 155, 150, 130, 135])
print(izm2 - izm1)

stats.wilcoxon(izm1, izm2)

[-10  -5 -15 -15 -20]


WilcoxonResult(statistic=0.0, pvalue=0.0625)

Примем значение уровня значимости $\alpha=0.05$\
Т.к. p-value=0.06 больше 0.05, то гипотезу $H_0$ принимаем. Статистически значимой разницы между выборками нет.

***
## Задание 4.

### Даны 3 группы  учеников плавания. В 1 группе время на дистанцию 50 м составляют: 56, 60, 62, 55, 71, 67, 59, 58, 64, 67. Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53. Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54.

Поскольку исследуются три зависимые выборки, то выбираем критерий Крускала–Уоллиса.

In [5]:
gr1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
gr2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
gr3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])

stats.kruskal(gr1, gr2, gr3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

Если принять значение уровня значимости $\alpha=0.05$\
Т.к. p-value=0.07 больше 0.05, то гипотезу $H_0$ принимаем. Статистически значимой разницы между выборками нет.

***
## Задание 5.

### Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%.
2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

$H_0: a = a_0 = 2.5$

Выборочная средняя $a_1 = 2.53$, что больше предполагаемой генеральной средней $a_0=2.5$, потому проверим альтернативную гипотезу $H_1: a > a_0$ (правосторонняя критическая область).

Т.к. генеральная дисперсия $\sigma^2$ неизвестна используем t-критенрий (распределение Стьюдента): \
$t = \cfrac{(\bar{x}-a_0)\sqrt{n}}{S}$

In [6]:
a0, n, alpha = 2.5, 10, 0.05
X = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
a1 = np.mean(X)
a1

2.5279999999999996

In [7]:
Tk = stats.t.ppf(1 - alpha, n-1)
Tk

1.8331129326536335

In [8]:
sd = np.std(X, ddof=1)
sd

0.1572542173961923

In [9]:
Tn = (a1 - a0) * np.sqrt(n) / sd
Tn

0.563061366180296

In [10]:
print(f'H0: a = 2.5 is {Tn<Tk}')

H0: a = 2.5 is True


Полученное значение $t_н = 0.563$ попало в область принятия гипотезы $(t<1.833)$, поэтому на уровне значимости $0,05$ нулевую гипотезу $H_0: a = 2.5$ принимаем.